WooCommerce/Classes/ViewRelated/Products/Variations/ProductVariationsViewController.swift

```diff
@@ -179,6 +213,22 @@ private extension ProductVariationsViewController {
 
         tableView.reloadData()
     }
+
+    func configureResultsControllerEventHandling(_ resultsController: ResultsController<StorageProductVariation>) {
+        let onReload = { [weak self] in
+            self?.tableView.reloadData()
+            self?.updateTopBannerView()
+        }
+
+        resultsController.onDidChangeContent = { [weak tableView] in
+            tableView?.endUpdates()
+            onReload()
+        }
+
+        resultsController.onDidResetContent = {
+            onReload()
+        }
+    }
```

Comment:
This looks good, but it got me thinking about what would happen if there were a lot of variations. In that case, the banner would not appear immediately if the first product without the price was beyond page 25, and it would probably also cause a small graphic glitch since the tableview offset would change. Is it right? Or maybe we already know the price for all the variations?

Reply:
There's no other API for us to know if a product has variations without a price, we can only tell after fetching variations from the API in pages. Yea it is possible where the variations without a price are not in the first page of results, so the banner could appear after fetching a later page. Here's a gif of a product with 71 variations with the last row as the only variation without a price (feel free to check out the Woo BBQ product on my test site as well):

![simulator](https://user-images.githubusercontent.com/1945542/90599108-dcd80280-e226-11ea-97ac-16c8503e7c34.gif)

I didn't find the offset too jarring, the next page was triggered when viewing the bottom of the results so I felt rows coming in was natural. Lemme know if you think another way to implement the header would be a better UX though!

Reply:
Sounds good, don't worry @jaclync. I think that at the moment with the API at our disposal we have only two ways of improving it:
1. fetch all the variants (iterating through all the pages), but it's not ideal.
2. show the banner always fixed on top, and it will be correct to hide it if the page doesn't contain variants without price. What do you think about this @Garance91540?

Reply:
I generally avoid fetching all the pages if not really necessary, and I think in this case of warning banner it's fine if it doesn't show up until later pages are fetched.

If a user is looking for a variation to edit, I think they will scroll down to find the exact variation and the row also shows "No price" in warning color in addition to the warning UI in variation details. Also, I don't have the data yet but I'd think the majority of active products have fewer than 25 variations (it feels hard to manage more than 25 individually 🙂 ).